BootloaderCorePkg/Tools/GenContainer.py

```diff
@@ -736,7 +745,6 @@ def main():
     cmd_display.add_argument('-c',  dest='compress', choices=['lz4', 'lzma', 'dummy'],  default='dummy', help='compression algorithm')
     cmd_display.add_argument('-a',  dest='auth', choices=['SHA2_256', 'SHA2_384', 'RSA2048', 'RSA3072' 'none'], default='none',  help='authentication algorithm')
     cmd_display.add_argument('-k',  dest='key_file',  type=str, default='', help='Private key file path to sign component')
-    cmd_display.add_argument('-ht', dest='hash_type', type=str,  choices=['SHA2_256', 'SHA2_384'], default='SHA2_256', help='Signing Hash Alg when auth type is RSA2048,RSA3072')

```

Comment:
Why do we remove all hash_type?  
There will be case that the component will not be signed individually.  Instead, it just does hash verification. In this case, there is no public key at all.  How do we know the hash type to use ?

Reply:
Agreed with comments. There were two cases,
- Hash type used in container generation.
- Hash type used in signing mechanism
By removal of hash_type param, it would generate error in cases where this tool used in standalone mode and private key is NULL.

I would update with error check for validating hash type to be used when private key is valid.

Reply:
For hash verification only,  'auth' variable will list that.

Reply:
if we can converge to use auth to cover both cases,  that will be fine.   Then we can remove hash_type.